src/langchain_google_spanner/loader.py

```diff
@@ -0,0 +1,422 @@
+            content_columns: The list of column(s) or field(s) to use for a Document's page content.
+                              Page content is the default field for embeddings generation.
+            metadata_columns: The list of column(s) or field(s) to use for metadata.
+            format: Set the format of page content if using multiple columns or fields.
+                    Format included: 'text', 'JSON', 'YAML', 'CSV'.
+            databoost: Use data boost on read. Note: needs extra IAM permissions and higher cost.
+            metadata_json_column: The name of the JSON column to use as the metadata's base dictionary.
+            staleness: The time bound for stale read. Takes either a datetime or float.
+            client: The connection object to use. This can be used to customize project id and credentials.
+        """
+        self.instance_id = instance_id
+        self.database_id = database_id
+        self.query = query
+        self.content_columns = content_columns
+        self.metadata_columns = metadata_columns
+        self.format = format
+        self.metadata_json_column = metadata_json_column
+        formats = ["JSON", "text", "YAML", "CSV"]
+        if self.format not in formats:
+            raise Exception("Use one of 'text', 'JSON', 'YAML', 'CSV'.")
+        self.databoost = databoost
+        self.client = client_with_user_agent(client, USER_AGENT_LOADER)
+        self.staleness = staleness
+        instance = self.client.instance(instance_id)
+        if not instance.exists():
+            raise Exception("Instance doesn't exist.")
+        database = instance.database(database_id)
+        if not database.exists():
+            raise Exception("Database doesn't exist.")
+
+    def load(self) -> List[Document]:
+        """
+        Load langchain documents from a Spanner database.
+
+        Returns:
+            (List[langchain_core.documents.Document]): a list of Documents with metadata
+            from specific columns.
+        """
+        return list(self.lazy_load())
+
+    def lazy_load(self) -> Iterator[Document]:
+        """
+        A lazy loader for langchain documents from a Spanner database. Use lazy load to avoid
+        caching all documents in memory at once.
+
+        Returns:
+            (Iterator[langchain_core.documents.Document]): a list of Documents with metadata
+            from specific columns.
+        """
+        instance = self.client.instance(self.instance_id)
+        db = instance.database(self.database_id)
+        timestamp = (
+            self.staleness.replace(tzinfo=datetime.timezone.utc)
+            if type(self.staleness) is datetime.datetime
+            else None
+        )
+        duration = (
+            datetime.timedelta(seconds=self.staleness)
+            if type(self.staleness) is float
+            else None
+        )
+        snapshot = db.batch_snapshot(exact_staleness=duration, read_timestamp=timestamp)
```

Comment:
We should support all the four options: 

-  read_timestamp: Optional[datetime.datetime] = None,
-  min_read_timestamp: Optional[datetime.datetime] = None,
-  max_staleness: Optional[datetime.timedelta] = None,
-  exact_staleness: Optional[datetime.timedelta] = None,

Reply:
The design was scoped to just read_timestamp and exact_staleness